temba/flows/models.py

```diff
@@ -670,7 +669,9 @@ def handle_actionset(cls, actionset, step, run, msg, started_flows, is_test_cont
 
     @classmethod
     def handle_ruleset(cls, ruleset, step, run, msg, started_flows, resume_parent_run=False, resume_after_timeout=False):
-        if msg.status == INTERRUPTED:  # check interrupt
+        from temba.ussd.models import USSDSession
+
+        if ruleset.is_ussd() and run.session and run.session.status == USSDSession.INTERRUPTED:
```

Comment:
session interrupted is for dealing with "interrupted" ruleset options, so if there are other actions  connected the run will still proceed with them. another example would be to restart the session after an interrupt "signal" happened (this is for later when we actually have gateways that support it)